server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -719,26 +731,50 @@ public void dispose() {
      * @param attr Serializable of the entire attributes collection
      * @since 2.2.0
      */
-    public void setAttributesRaw(HashMap<String, Object> attr) {
-        this.attributes = (attr == null) ? new HashMap<>() : attr;
+    public void setAttributesRaw(Map<String, Object> attr) {
+        this.attributes = attr;
+    }
+
+    @Override
+    public Stream<Attribute> attributes() {
+        return this.attributes.entrySet().stream().map(entry -> new Attribute(AttributeName.of(entry.getKey()), AttributeValue.ofAny(entry.getValue())));
     }
 
     @Override
     public Serializable getAttribute(String key) {
         return (Serializable) attributes.get(key);
     }
 
+    @Override
+    public Optional<Attribute> getAttribute(AttributeName name) {
+        return Optional.ofNullable(attributes.get(name.asString())).map(value -> new Attribute(name, AttributeValue.ofAny(value)));
```

Comment:
`name.asString()` is not safe, it could throws NullPointerException. Especially this is a public method, it could be invoked from everywhere and we could not control the null safe guarantee from all of it's clients.

Reply:
AttributeName is strongly typed and we can assume we can safely call `asString`

Or this object needs strong re-design...